Source/Scene/Expression.js

```diff
@@ -605,77 +666,129 @@ define([
     }
 
     // PERFORMANCE_IDEA: Determine if parent property needs to be computed before runtime
-    Node.prototype._evaluateMemberDot = function(feature, result) {
+    Node.prototype._evaluateMemberDot = function(feature) {
         if(checkFeature(this._left)) {
             return feature.getProperty(this._right);
         }
-        var property = this._left.evaluate(feature, result);
+        var property = this._left.evaluate(feature);
         if (!defined(property)) {
             return undefined;
         }
         return property[this._right];
     };
 
-    Node.prototype._evaluateMemberBrackets = function(feature, result) {
+    Node.prototype._evaluateMemberBrackets = function(feature) {
         if(checkFeature(this._left)) {
-            return feature.getProperty(this._right.evaluate(feature, result));
+            return feature.getProperty(this._right.evaluate(feature));
         }
-        var property = this._left.evaluate(feature, result);
+        var property = this._left.evaluate(feature);
         if (!defined(property)) {
             return undefined;
         }
-        return property[this._right.evaluate(feature, result)];
+        return property[this._right.evaluate(feature)];
     };
 
-    Node.prototype._evaluateArray = function(feature, result) {
+    Node.prototype._evaluateArray = function(feature) {
         var array = [];
-        for (var i = 0; i<this._value.length; i++) {
-            array[i] = this._value[i].evaluate(feature, result);
+        for (var i = 0; i < this._value.length; i++) {
+            array[i] = this._value[i].evaluate(feature);
         }
         return array;
     };
 
     // PERFORMANCE_IDEA: Have "fast path" functions that deal only with specific types
     // that we can assign if we know the types before runtime
 
-    Node.prototype._evaluateNot = function(feature, result) {
-        return !(this._left.evaluate(feature, result));
+    Node.prototype._evaluateNot = function(feature) {
+        var left = this._left.evaluate(feature);
+
+        //>>includeStart('debug', pragmas.debug);
+        if (typeof(left) !== 'boolean') {
```

Comment:
Are you sure we want all these `if` checks for types?  The mantra is to rely on validation for comprehensive testing and then to have runtime checks only for common critical issues.  Perhaps this falls into the later category?